### DISCUSSION TITLE
swayidle doesn't work

### DISCUSSION THREAD

**Discussion by austinweisgrau** [regolith-linux/regolith-desktop#967] [Q&A] [answered]
I'm running regolith 3.0 on Wayland on Ubuntu 22.04 LTS

swayidle seems to be totally defunct. I can't get it to notice an idle period.

Here is the output for a sample command from the commandline:
```
$ swayidle -d timeout 5 'swaylock'
Got timeout
Register idle timeout at 5000 ms
Setup idle
Command: swaylock
No config file found.
Using session: /org/freedesktop/login1/session/_37
Enable idle timeouts
Register with timeout: 5000
```

After that, nothing is registered, even if the system is left to idle for a long time, the swaylock doesn't begin, etc.

I'm new to wayland and digging in at this level so not sure how to diagnose at a deeper level what is going on here.

The closest discussion I could find is [here](https://github.com/orgs/regolith-linux/discussions/876#discussioncomment-6715055), which says that swayidle configuration is controlled by the regolith-control-center Power menu. Perhaps the implication is that swayidle is completely inaccessible outside of that utility. Ideally, I would like to be able to customize behavior on resume with custom commands, etc.

Thank you for any help!

**Comment by austinweisgrau** [comment]
Today I ran "sudo apt update" and "sudp apt install sway" and swayidle is now working!

However, separate issue, apt is considering most of regolith sway to be deserving of removal, not sure what's going on there. Will make a separate issue for that.

**Reply by kgilmer** [reply]
@SoumyaRanjanPatnaik might there be a missing dependency for `swayidle`?

@austinweisgrau Regolith has an unofficial fork of Sway called `sway-regolith`.   You can explicitly install it with apt to keep it from being removed.  We do not use the package `sway` FWIW.

**Reply by SoumyaRanjanPatnaik** [reply]
My bad for initially misreading this issue. Regolith uses `regolith-powerd` for managing idle settings. It maintains its own instance of sway idle and keeps it in sync with the idle settings in the `gnome-control-center`. 

When you installed `sway`, `sway-regolith` and consequently `regolith-powerd` must have been removed, which is why `swayidle`works for you now. Like @kgilmer said, `sway-regolith` is a fork of sway, and also a hard dependency for most of the`sway` session of Regolith.

**Reply by haaspors** [reply]
@SoumyaRanjanPatnaik Could you point me to where and how regolith is starting `regolith-powerd` when running sway?

**Comment by SoumyaRanjanPatnaik** [comment, marked as answer]
Regolith uses `regolith-powerd` for managing idle settings. It maintains its own instance of `swayidle` and keeps it in sync with the idle settings in the gnome-control-center. In order to maintain your own `swayidle` configuration, you need to remove the `regolith-powerd` package.

**Comment by austinweisgrau** [comment]
Got it, that all makes a lot of sense, thank you so much for the clarifications!

**Comment by pmahnke** [comment]
I am having issues with my system never suspending, or only after I open the lid.  I notice that when I try to run `systemctl suspend` I get this error:

```
systemctl suspend
Operation inhibited by "peter" (PID 3973 "gnome-session-b", user peter), reason is "user session inhibited".
``` 

should the command be run with the "-i" flag? or is something else happening?

**Reply by SoumyaRanjanPatnaik** [reply]
Few questions:

Are you on the sway session or the i3 session? 
How frequently does it happen? 
Do you have any open apps that might prevent the system from idling? 
Do you usually have audio playing in the background when this happens?

**Reply by pmahnke** [reply]
_Are you on the sway session or the i3 session?_

```
> sway-regolith --version
sway version 1.8
```

_How frequently does it happen?_

100% of the time

_Do you have any open apps that might prevent the system from idling?_

I run: joplin, simplenote, chrome and wezterm mostly  I don't know if a specific site in chrome could stop it.  I have also tested quitting joplin and that didn't change things.

_Do you usually have audio playing in the background when this happens?_

no; however, I often am on nordvpn... if that matters?


Thanks for helping with this, I have tried to understand it, but I cannot.

**Reply by pmahnke** [reply]
@SoumyaRanjanPatnaik any thoughts?

**Reply by SoumyaRanjanPatnaik** [reply]
What version of `gnome-session-binary` do you have? And does this also happen right after a startup?

**Reply by pmahnke** [reply]
@SoumyaRanjanPatnaik it says `45.0-1ubuntu1` does that make sense? It doesn't happen right after startup.

**Reply by SoumyaRanjanPatnaik** [reply]
Ohh. Can you figure out a way to reproduce this? Maybe there's a particular app that inhibits idle!!

**Reply by pmahnke** [reply]
I assume I can just leave my computer untouched and it will happen.  How can I trap which app is stopping it?
